src/org/authInfo.ts

```diff
@@ -1106,7 +1106,22 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
       this.logger.info(`Exchanging auth code for access token using loginUrl: ${options.loginUrl}`);
       authFields = await oauth2.requestToken(ensure(options.authCode));
     } catch (err) {
-      throw messages.createError('authCodeExchangeError', [(err as Error).message]);
+      let error: SfError;
+      let errorMsg: string;
+      if (err instanceof Error) {
+        errorMsg = `${err.name}::${err.message}`;
+        error = SfError.create({
+          message: errorMsg,
+          name: 'AuthCodeExchangeError',
+          cause: err,
+        });
+      } else {
+        error = SfError.wrap(err);
+        errorMsg = error.message;
+      }
+      error.message = messages.getMessage('authCodeExchangeError', [errorMsg]);
```

Comment:
this would overwrite the message set from 1114?

Reply:
Kind of.  I want the server's error name and error description to be part of the authCode exchange error message.